xapian-core/api/queryinternal.cc

```diff
@@ -865,98 +869,158 @@ Query::Internal::unserialise(const char ** p, const char * end,
 	    return new Xapian::Internal::QueryValueRange(slot, begin, end_);
 	}
 	case 0: {
-	    // Other operators
-	    //
-	    //   000ttttt where:
-	    //     ttttt -> encodes which OP_XXX
-	    switch (ch & 0x1f) {
-		case 0x00: // OP_INVALID
-		    return new Xapian::Internal::QueryInvalid();
-		case 0x0a: { // Edit distance
-		    Xapian::termcount max_expansion;
-		    if (!unpack_uint(p, end, &max_expansion) || end - *p < 2) {
-			throw SerialisationError("not enough data");
-		    }
-		    int flags = static_cast<unsigned char>(*(*p)++);
-		    op combiner = static_cast<op>(*(*p)++);
-		    unsigned edit_distance;
-		    size_t fixed_prefix_len;
-		    string pattern;
-		    if (!unpack_uint(p, end, &edit_distance) ||
-			!unpack_uint(p, end, &fixed_prefix_len) ||
-			!unpack_string(p, end, pattern)) {
-			throw SerialisationError("not enough data");
-		    }
-		    using Xapian::Internal::QueryEditDistance;
-		    return new QueryEditDistance(pattern,
-						 max_expansion,
-						 flags,
-						 combiner,
-						 edit_distance,
-						 fixed_prefix_len);
-		}
-		case 0x0b: { // Wildcard
-		    Xapian::termcount max_expansion;
-		    if (!unpack_uint(p, end, &max_expansion) || end - *p < 2) {
-			throw SerialisationError("not enough data");
-		    }
-		    int flags = static_cast<unsigned char>(*(*p)++);
-		    op combiner = static_cast<op>(*(*p)++);
-		    string pattern;
-		    if (!unpack_string(p, end, pattern)) {
-			throw SerialisationError("not enough data");
-		    }
-		    return new Xapian::Internal::QueryWildcard(pattern,
-							       max_expansion,
-							       flags,
-							       combiner);
+		switch (ch >> 4)
```

Comment:
The formatting of this hunk of the patch is wrong in various ways - e.g. the `{` should be on the same line as the `switch`, some lines are overindented, some blocks aren't indented compare to their parent.

As James said on IRC, you really want to get the formatting correct when you write the code.  It's no fun for reviewers to have to point out glaring problems, and no fun for you to have to go through fixing them all.